Assets/Code/Scriptable Variables/AvatarPrefabMapping.cs

```diff
@@ -21,7 +21,9 @@ public class AvatarPrefabMapping : ScriptableVariable {
 
     //Get the first prefab that matches the supplied Gender. Null if one is not defined.
     public GameObject GetPrefab(UserGender gender) {
-      var mapping = _prefabs.Find(x => x._gender == gender);
+      var mapping = _prefabs.Find(x => x._gender == gender && x._isTaken == false);
+      int idx = _prefabs.FindIndex(x => x._prefab.name == mapping._prefab.name);
+      _prefabs[idx]._isTaken = true;
       return mapping?._prefab;
```

Comment:
What's going to happen here if all the prefabs are isTaken == true?  What we probably want is to continue cycling through the list. The scenario will have duplicate avatars in the scene, but that is better than no avatars.

There is actually a subtle issue here. You are modifying a ScriptableObject that auto-serializes -- it actually saves your runtime changes to file while Playing! You can see this by using the Unity Inspector to view the AvatarPrefabMapping .asset as you play. When you stop Playing, the .asset still contains your changes.

You can fix this by either:
1) don't modify the mapping scriptable object during runtime -- move your "isTaken" to outside the AvatarPrefabMapping class, or you leave it _inside_ the class and mark it with a `[NonSerialized]` attribute maybe.
2) "reset" the scriptableObject every time the game is started and ended. Maybe the `UserFactory` would call a reset function on the AvatarPrefabMapping during its `Awake()` and `OnDestroy()`